app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddPathSurfaceForm.kt

```diff
@@ -1,56 +1,43 @@
+            Item("unpaved", R.drawable.path_surface_unpaved, R.string.quest_surface_value_unpaved, null, listOf()) +
+            Item("ground", R.drawable.surface_ground, R.string.quest_surface_value_ground, null, listOf())
 
-    override val topItems get() =
-        when (val pathType = determinePathType(osmElement!!.tags)) {
-            "bridleway" -> listOf(
-                DIRT, GRASS, SAND,
-                PEBBLES, FINE_GRAVEL, COMPACTED
-            )
-            "path" -> listOf(
-                DIRT, PEBBLES, COMPACTED,
-                ASPHALT, FINE_GRAVEL, PAVING_STONES
-            )
-            "footway" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                COMPACTED, FINE_GRAVEL, DIRT
-            )
-            "cycleway" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                COMPACTED, WOOD, METAL
-            )
-            "steps" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                WOOD, SETT, UNHEWN_COBBLESTONE
-            )
-            else -> throw IllegalStateException("Unexpected path type $pathType")
-        }.toItems()
+    override val itemsPerRow = 3
 
-    override val allItems = listOf(
-        // except for different panorama images, should be the same as for the road quest, to avoid confusion
-        Item("paved", R.drawable.panorama_path_surface_paved, R.string.quest_surface_value_paved, null, PAVED_SURFACES.toItems()),
-        Item("unpaved", R.drawable.panorama_path_surface_unpaved, R.string.quest_surface_value_unpaved, null, UNPAVED_SURFACES.toItems()),
-        Item("ground", R.drawable.panorama_surface_ground, R.string.quest_surface_value_ground, null, GROUND_SURFACES.toItems())
-    )
-
-    private fun determinePathType(tags: Map<String, String>): String? {
-        val pathType = tags["highway"]
-        // interpret paths with foot/bicycle/horse=designated as...
-        if ("path" == pathType) {
-            if ("designated" == tags["bicycle"]) return "cycleway"
-            if ("designated" == tags["horse"]) return "bridleway"
-            if ("designated" == tags["foot"]) return "footway"
+    override fun onClickOk(selectedItems: List<String>) {
+        val value = selectedItems.single()
+        if(value == "paved" || value == "unpaved" || value == "ground") {
+            AlertDialog.Builder(requireContext())
+                .setMessage(R.string.quest_surface_detailed_answer_impossible_confirmation)
+                .setPositiveButton(R.string.quest_generic_confirmation_yes) {
+                    _, _ -> run {
+                        DescribeGenericSurfaceDialog(requireContext()) { description ->
+                          applyAnswer(DetailingWhyOnlyGeneric(value, description))
+                        }.show()
+                    }
+                }
+                .setNegativeButton(android.R.string.cancel, null)
+                .show()
```

Comment:
I think this UI could be improved here: This could go into one dialog, a la "Can you really not specify it more detailed? Then please describe the surface here, for example...."

Reply:
I worried about fitting text and field on one screen - especially on small screens and in languages where the same text is much longer than in English.

Also, there is a plenty of text (though maybe it can be shortened) so I thought that in such layout there is a greater chance that it will be actually noticed and read.

But if you think that it should be changed it should fairly simple to do - let me know if I should do this.

Reply:
It's a dialog, it could potentially cover most of the screen. I don't think there is really not enough space here

Reply:
![screen04](https://user-images.githubusercontent.com/899988/96624986-da208900-130d-11eb-84ce-0f955edd9a5a.jpg)

vs text extracted into extra dialog

![screen05](https://user-images.githubusercontent.com/899988/96625042-ec9ac280-130d-11eb-90b3-1d744572be14.jpg)

I will implement merged version but I think that one extra screen is better.

Or maybe this text can be shortened solving root problem?

Reply:
Done

Reply:
Ok, this is indeed a long text. I didn't realize you mention 1. differs along the way, 2. can't say 3. 255 characters, 4. an example.

You are right, in this case maybe a second dialog is indeed better.

Reply:
Done